server/resourcemanager/resource_manager_test.go

```diff
@@ -0,0 +1,197 @@
+package resourcemanager_test
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"testing"
+
+	"github.com/gorilla/mux"
+	"github.com/kubeshop/tracetest/server/id"
+	"github.com/kubeshop/tracetest/server/resourcemanager"
+	rmtests "github.com/kubeshop/tracetest/server/resourcemanager/testutil"
+	"github.com/stretchr/testify/mock"
+)
+
+type sampleResource struct {
+	ID   id.ID  `mapstructure:"id"`
+	Name string `mapstructure:"name"`
+
+	SomeValue string `mapstructure:"some_value"`
+}
+
+func (sr sampleResource) HasID() bool {
+	return sr.ID.String() != ""
+}
+
+func (sr sampleResource) Validate() error {
+	return nil
+}
+
+type sampleResourceManager struct {
+	mock.Mock
+}
+
+func (m *sampleResourceManager) SetID(sr sampleResource, id id.ID) sampleResource {
+	sr.ID = id
+	return sr
+}
+
+func (m *sampleResourceManager) Create(_ context.Context, s sampleResource) (sampleResource, error) {
+	args := m.Called(s)
+	return args.Get(0).(sampleResource), args.Error(1)
+}
+
+func (m *sampleResourceManager) Update(_ context.Context, s sampleResource) (sampleResource, error) {
+	args := m.Called(s)
+	return args.Get(0).(sampleResource), args.Error(1)
+}
+
+func (m *sampleResourceManager) Get(_ context.Context, id id.ID) (sampleResource, error) {
+	args := m.Called(id)
+	return args.Get(0).(sampleResource), args.Error(1)
+}
+
+func (m *sampleResourceManager) Delete(_ context.Context, id id.ID) error {
+	args := m.Called(id)
+	return args.Error(0)
+}
+
+func (m *sampleResourceManager) List(_ context.Context, take, skip int, query, sortBy, sortDirection string) ([]sampleResource, error) {
+	args := m.Called(take, skip, query, sortBy, sortDirection)
+	return args.Get(0).([]sampleResource), args.Error(1)
+}
+func (m *sampleResourceManager) Count(_ context.Context, query string) (int, error) {
+	args := m.Called(query)
+	return args.Int(0), args.Error(1)
+}
+
+func TestSampleResource(t *testing.T) {
+
+	sample := sampleResource{
+		ID:        "1",
+		Name:      "the name",
+		SomeValue: "the value",
+	}
+
+	sampleUpdated := sampleResource{
+		ID:        "1",
+		Name:      "the name updated",
+		SomeValue: "the value updated",
+	}
+
+	rmtests.TestResourceTypeWithErrorOperations(t, rmtests.ResourceTypeTest{
```

Comment:
This test serves to test the actual resource manager. It relies on a mock instead of an actual DB so we can have better control of what happens at each step. 

For example, we can easily simulate DB errors to assert they are handled correctly and nicely presented to users. 

It uses the `TestResourceTypeWithErrorOperations` function (instead of `TestResourceType`) because it includes operations targeted at checking these kinds of internal error handling.

These generic/shared functionalities are tested here, leaving each resource implementation with the responsibility to test its internal workings by thenselves.